docs/concepts/architecture.md

```diff
@@ -113,3 +113,43 @@ Nuclio provides the ability to run different versions of the same function simul
 
 The event-source mapping can either specify the exact function version, or utilize a defined alias and thus eliminate the need to change the mapping when a newer version of the function is published.
 
+
+## Processors' internal abstractions
+
+Here is an illustration of the abstraction levels that every event goes through while being processed:
+![event-processing-flow](../../docs/assets/images/internal-processor-abstractions.png)
+
+Thus, every event goes through two EventProcessor levels: the first at the trigger level and the second at the runtime level.
+The EventProcessor exists to generalize the allocation logic and make it abstract.
```

Comment:
The EventProcessor interface generalizes the allocation logic, and shares it between different resources.